### PR TITLE
allow save mesh with an URI

### PR DESCRIPTION
**Pull request by vcloarec**
This PR allows to save mesh with an URI.

### CODE REVIEW

File: tests/mdal_testutils.cpp [hunk 1]
```diff
@@ -383,14 +384,16 @@ void saveAndCompareMesh( const std::string &filename, const std::string &savedFi
...
+  std::string savedUri = MDAL::buildMeshUri( savedFile, meshName, driver );
+
   // Open mesh
   MDAL_MeshH meshToSave = MDAL_LoadMesh( uri.c_str() );
   EXPECT_NE( meshToSave, nullptr );
   MDAL_Status s = MDAL_LastStatus();
   ASSERT_EQ( MDAL_Status::None, s );
 
   // Save the mesh
-  MDAL_SaveMesh( meshToSave, savedFile.c_str(), driver.c_str() );
```
Comment (PeterPetrik) [inline tests/mdal_testutils.cpp:393]: can we keep boths MDAL_SaveMesh and MDAL_SaveMeshWithURi in the test to test both of them (both are still in public API)

File: mdal/api/mdal.h [hunk 2]
```diff
@@ -273,6 +273,16 @@ MDAL_EXPORT MDAL_MeshH MDAL_CreateMesh( MDAL_DriverH driver );
  */
 MDAL_EXPORT void MDAL_SaveMesh( MDAL_MeshH mesh, const char *meshFile, const char *driver );
 
+/**
+ * Saves mesh (only mesh structure) on a file with an uri. On error see MDAL_LastStatus for error type.
+ *
+ * uri has form <DriverName>:"<MeshFilePath>"[:<SpecificMeshName>]
+ * examples: Ugrid:"mesh.nc":0, Ugrid:"mesh.nc":mesh2d, Ugrid:"mesh.nc"
+ *
```
Comment (PeterPetrik) [inline mdal/api/mdal.h:281]: - is `SpecificMeshName` used in any driver?

**Review by PeterPetrik — APPROVED** [review]

File: mdal/api/mdal.h (hunk 2 above)
Comment (vcloarec) [reply mdal/api/mdal.h:281]: not for now..., but UGRID has "mesh1D","mesh2D". Later, if UGRID support writing 1D, we will need that

File: tests/mdal_testutils.cpp (hunk 1 above)
Comment (vcloarec) [reply tests/mdal_testutils.cpp:393]: done

**Pull request merged**